Assignment5/miguelventoso - 21A284/assignment5_miguelventoso.py

```diff
@@ -0,0 +1,316 @@
+                  output_path=output_path,
+                  project_name=project_name)
+
+"""The KGTK setup command defines environment variables for all the files so that you can reuse the Jupyter notebook when you install it on your local machine."""
+
+ck.print_env_variables()
+
+# Commented out IPython magic to ensure Python compatibility.
+# %%time
+# ck.load_files_into_cache()
+
+"""# About this assignment.
+This assignment is based on https://github.com/usc-isi-i2/kgtk-notebooks/tree/main/tutorial. If you have any questions or doubts, it is encouraged to look how the tutorial performs the different operations.
+
+Additional information can be found in https://kgtk.readthedocs.io/
+
+## Simple graph statistics
+
+Let's calculate first some statistics about the KG. Count the number of instances:
+"""
+
+# Commented out IPython magic to ensure Python compatibility.
+# %%time
+# #P31 is the predicate "instance of"
+# print(kgtk("""
+#     query -i all
+#     --match '(instance)-[:P31]->()'
+#     --return 'count(distinct instance) as n_instances'
+# """))
+
+"""Now, count the number of distinct properties: 
+
+"""
+
+# Commented out IPython magic to ensure Python compatibility.
+# %%time
+# #Output with print to avoid exception 'NotebookFormatter' object has no attribute 'get_result'
+# print(kgtk("""
+#     query -i all
+#     --match '()-[{label: property}]->()'
+#     --return 'count(distinct property) as n_properties'
+# """))
+
+"""Now, let's count the frequency of those properties. That is, how many instances we can find with each property"""
+
+# Commented out IPython magic to ensure Python compatibility.
+# %%time
+# print(kgtk("""
+#     query -i all
+#     --match '(instance)-[p {label: property}]->()'
+#     --return 'p.label, count(distinct instance) as frequency'
+# """))
+
+"""## Simple queries
+Some of these queries are simple and will run in the Wikidata endpoint. 
+Try both of them using SPARQL and Kypher
+"""
+
+# Which actors has Schwarzenegger worked with throughout his career? (Print also the movie)
+
+# (in SPARQL)
+# TO DO 
+'''
+SELECT ?actor ?movie ?movieLabel ?actorLabel WHERE {
+   ?movie wdt:P161 wd:Q2685.
+   ?movie wdt:P161 ?actor.
+   SERVICE wikibase:label { 
+     bd:serviceParam wikibase:language "[AUTO_LANGUAGE],en". 
+   }
+   FILTER (?actor != wd:Q2685)
+}
+'''
+#Link to the result: https://w.wiki/4Pp6
+
+
+# In Kypher:
+print(kgtk("""
+    query -i all
+    --match '(:Q2685)<-[:P161]-(movie)-[:P161]->(actor)'
+    --where 'actor != "Q2685"'
+     --return 'actor as actor, movie as movie'
+    / add-labels
+"""))
+
+# How many awards does Schwarzenegger have?
+
+# SPARQL:
+# TO DO
+'''
+SELECT (
+  COUNT(DISTINCT ?awards) as ?num_awards
+)  
+WHERE {
+  wd:Q2685 wdt:P166 ?awards.
+}
+'''
+
+#Link to result: https://w.wiki/4Pp8
+
+# Kypher:
+print(kgtk("""
+    query -i all 
+    --match '(:Q2685)-[:P166]->(awards)' 
+    --return 'count(distinct awards) as num_awards' 
```

Comment:
Note that by counting distinct awards, if the actor won the same award twice, it will only be counted once.